src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+      notifications: new NotificationsApi(v1Config, undefined, axiosInstance),
+      reactions: new ReactionsApi(v1Config, undefined, axiosInstance),
+      v1Follows: new FollowsApiV1(v1Config, undefined, axiosInstance),
+      v2Cast: new CastApiV2(v2Config, undefined, axiosInstance),
+      signer: new SignerApi(v2Config, undefined, axiosInstance),
+      reaction: new ReactionApi(v2Config, undefined, axiosInstance),
+      v2Follow: new FollowApiV2(v2Config, undefined, axiosInstance),
+    };
+  }
+
+  /**
+   * Utility for parsing errors returned by the Merkle API server. Returns true
+   * if the given error is caused by an error response from the server, and
+   * narrows the type of `error` accordingly.
+   */
+  public static isApiErrorResponse(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    error: any
+  ): error is WithRequired<AxiosError<ErrorRes>, "response"> {
+    if (!(error instanceof AxiosError)) return false;
+    return error.response?.data !== undefined;
+  }
+
+  /**
+   * Creates a Signer. See [Neynar documentation](https://docs.neynar.com/reference/create-signer)
+   * for more details.
+   */
+  public async createSigner(): Promise<Signer> {
+    const response = await this.apis.signer.createSigner();
+    return response.data;
+  }
+
+  /**
+   * Fetches an existing Signer. See [Neynar documentation](https://docs.neynar.com/reference/get-signer)
+   * for more details.
+   *
+   */
+  public async fetchSigner(signerUuid: string): Promise<Signer | undefined> {
+    try {
+      const response = await this.apis.signer.signer(signerUuid);
+      return response.data;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        const status = error.response.status;
+        if (status === 404) {
+          return undefined;
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Registers a Signer with an fid. See [Neynar documentation](https://docs.neynar.com/reference/register-app-fid)
+   * for more details.
+   */
+  public async registerSigner(
+    signerUuid: string,
+    fid: number,
+    deadline: number,
+    signature: string
+  ): Promise<Signer> {
+    const body: RegisterSignerKeyReqBody = {
+      signer_uuid: signerUuid,
+      app_fid: fid,
+      deadline,
+      signature,
+    };
+    const response = await this.apis.signer.registerSignedKey(body);
+    return response.data;
+  }
+
+  /**
+   * Gets information about an individual cast using neynar v1 API
+   */
+  public async v1FetchCast(
+    castOrCastHash: v1Cast | string
+  ): Promise<v1Cast | undefined> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const response = await this.apis.v1Cast.cast(castHash);
+    if (response.status === 404) return undefined;
+    return response.data.result.cast ?? undefined;
```

Comment:
Why would this be unset?

Reply:
only in the case of a 404, and in the updated code it will return null.